core/src/main/java/com/lantanagroup/link/FhirDataProvider.java

```
@@ -342,27 +346,36 @@ public Bundle searchBundleByTag(String system, String value) {
             .execute();
   }
 
-  public void submitToServer(String serverBase, FHIRSenderOAuthConfig senderConfig, Resource resource) {
+  public void submitToServer(String serverBase, String token, Resource resource) {
     IGenericClient submissionClient = this.ctx.newRestfulGenericClient(serverBase);
     submissionClient.registerInterceptor(new GZipContentInterceptor());
-    IClientInterceptor authInterceptor = new BasicAuthInterceptor(senderConfig.getUsername(), senderConfig.getPassword());
+    BearerTokenAuthInterceptor authInterceptor = null;
+    try {
+      authInterceptor = new BearerTokenAuthInterceptor(token);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
     submissionClient.registerInterceptor(authInterceptor);
```

This will throw a `NullPointerException` if the above constructor for `BearerTokenAuthInterceptor` failed.  Not sure why we're catching that exception.  Do we need to allow `token` to be null, and *not* add the interceptor in that case?  If so, I would recommend removing the `try`/`catch` and replacing it with an explicit null check for `token`.  Otherwise we should probably let the exception bubble up to the caller.